std/exception.d

```diff
@@ -1613,3 +1614,462 @@ version(unittest) package
+                @property auto back()
+                {
+                    try
+                    {
+                        return this.range.back;
+                    }
+                    catch(E exception)
+                    {
+                        return handler(exception, this.range);
+                    }
+                }
+            }
+            else
+            {
+                @property auto back()
+                {
+                    return this.range.back;
+                }
+            }
+
+            static if (functionsToHandle & RangePrimitive.popBack)
+            {
+                void popBack()
+                {
+                    try
+                    {
+                        return this.range.popBack();
+                    }
+                    catch(E exception)
+                    {
+                        handler(exception, this.range);
+                    }
+                }
+            }
+            else
+            {
+                void popBack()
+                {
+                    this.range.popBack();
+                }
+            }
+        }
+
+        static if (isRandomAccessRange!IRange)
+        {
+            static if (is(typeof(range[ulong.max])))
+                private alias opIndex_t = ulong;
+            else
+                private alias opIndex_t = uint;
+
+            static if (functionsToHandle & RangePrimitive.opIndex)
+            {
+                auto ref opIndex(opIndex_t index)
+                {
+                    try
+                    {
+                        return this.range[index];
+                    }
+                    catch(E exception)
+                    {
+                        return handler(exception, this.range);
+                    }
+                }
+            }
+            else
+            {
+                auto ref opIndex(opIndex_t index)
+                {
+                    return range[index];
+                }
+            }
+        }
+
+        static if (hasLength!IRange)
+        {
+            static if (functionsToHandle & RangePrimitive.length)
+            {
+                @property auto length()
+                {
+                    try
+                    {
+                        return this.range.length;
+                    }
+                    catch(E exception)
+                    {
+                        return handler(exception, this.range);
+                    }
+                }
+
+            }
+            else
+            {
+                @property auto length()
+                {
+                    return this.range.length;
+                }
+            }
+
+            alias opDollar = length;
+        }
+
+        static if (hasSlicing!IRange)
+        {
+            static if (is(typeof(range[ulong.max .. ulong.max])))
+                private alias opSlice_t = ulong;
+            else
+                private alias opSlice_t = uint;
```

Comment:
Again, I think we should consider using `ParameterTypeTuple` to extract the exact argument types fo `opSlice` instead of assuming `ulong`/`uint`.